packages/scope-manager/tsconfig.spec.json

```diff
@@ -2,8 +2,9 @@
   "extends": "./tsconfig.json",
   "compilerOptions": {
     "outDir": "../../dist/out-tsc",
-    "module": "commonjs",
-    "types": ["jest", "node"]
+    "module": "Node16",
+    "types": ["jest", "node"],
+    "moduleResolution": "Node16"
```

Comment:
Hmm, why this change? Tests still work for me locally after reverting it.

Reply:
But does it typecheck fine without it?
There's a new error in this version to ensure that things are configured correctly.

Reply:
Yeah I like moving to the newer module & module resolution settings, just curious.

Also, per https://www.typescriptlang.org/tsconfig#moduleResolution, `moduleResolution` matches `module` by default with `node16` and other newer targets.

Reply:
Without this change, a type error will occur.

Reply:
Ah! https://devblogs.microsoft.com/typescript/announcing-typescript-5-2-rc/#module-and-moduleresolution-must-match-under-recent-node-js-settings

```plaintext
       tsconfig.spec.json:5:15 - error TS5110: Option 'module' must be set to 'Node16' when option 'moduleResolution' is set to 'Node16'.
       
       5     "module": "CommonJS",
                       ~~~~~~~~~~
```

Fun! In that case, I think we can get away with just the `"module": "Node16"`:

```suggestion
```

Reply:
Thank you! [e829631](https://github.com/typescript-eslint/typescript-eslint/pull/7451/commits/e82963179840354b8d091e452f2eff52dded63eb)